compiler/rustc_target/src/spec/targets/wasm32v1_none.rs

```diff
@@ -29,11 +29,10 @@ pub(crate) fn target() -> Target {
             // For now this target just never has an entry symbol no matter the output
             // type, so unconditionally pass this.
             "--no-entry",
-            // It is also necessary to explicitly set the features since some changes
+            // It is also necessary to explicitly set CPU since some changes
             // were made to WebAssembly starting with LLVM 20.1.0:
             // https://releases.llvm.org/20.1.0/docs/ReleaseNotes.html#changes-to-the-webassembly-backend
-            "--features=+mutable-globals",
-            "--no-check-features",
+            "--mllvm=-mcpu=mvp",
```

Comment:
Doing it at the place I indicated (the WasmLd implementation) has the advantage that it works for all wasm targets and also when the user passes `-Ctarget-cpu` to override the target default.

Reply:
Wasm-ld can't notice any difference between setting this cli arg in the target spec and setting it in the WasmLd implementation I linked to.

Reply:
Well I tried it locally and it really helped. It also puts these arguments in the output of `cargo build --verbose`.

Reply:
Can you show demo code with RUSTC_LOG=debug where -C target-feature won't work? It seemed ok to me. The only problem is that with an empty file in the target_features section the features enabled by the user won't be displayed (since .bc files don't save features).

Reply:
Yes, with a fallback to whatever target cpu and features is in the target spec. This is also what we do for gcc for example: https://github.com/rust-lang/rust/blob/b779120cad2c93e53c0dfb7007c1214b2b275183/compiler/rustc_codegen_ssa/src/back/linker.rs#L426 Actually maybe getting the target features right would be non-trivial as that isn't plumbed through yet and not used for gcc either. So maybe only set the target cpu in linker.rs and just add a FIXME for the target features (and if you do need to set the target features just do it in the target spec)